desc/equilibrium/equilibrium.py

```diff
@@ -698,8 +742,17 @@ def get_surface_at(self, rho=None, theta=None, zeta=None):
 
         if zeta is not None:
             assert (zeta >= 0) and (zeta <= 2 * np.pi)
-            surface = ZernikeRZToroidalSection(sym=self.sym, zeta=zeta)
+            warnif(
+                zeta != 0,
+                UserWarning,
+                "The lambda value of the returned surface will not be correct "
```

Comment:
why does this happen?

Reply:
I'd recommend doing the loop with R and Z, then we can get rid of `get_basis_poincare` entirely i think